### PR TITLE
feat: add option to configure tracked analytics

### PR DESCRIPTION
**Pull request by raae**
### Brief Summary of Changes

Add similar behavior to @cloudinary/url-gen for configuring tracked analytics. Work done on stream with @colbyfayock so he should also be able to answer questions about the code.

#### What Does This PR Address?
- [X] GitHub issue (Add reference - #590)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [x] Adds more tests

#### Are Tests Included?
- [X] Yes
- [ ] No

#### Reviewer, Please Note:
- View the description in #590
- Added one test but copied the expected tracking code from what it actually gave us, so make sure this is the expected tracking code
- ~Code changes only done in `lib`, not `lib-es5`~ now added
- Putting this as draft as I am not convinced we are done, but would still love a review

CleanShot 2023-02-21 at 21 49 23@2x

### CODE REVIEW

**Comment by colbyfayock** [comment]
Context: the DevX team would like @raae to add custom analytics to the Gatsby plugins that she's maintaining which is not possible

**Comment by raae** [comment]
To match the @cloudinary/url-gen we chose to add the option `trackedAnalytics` to be used as follows:

```js
  let imgStr = cloudinary.image("hello", {
      format: "png",
      trackedAnalytics: {
        sdkCode: "X",
        sdkSemver: "7.3.0",
        techVersion: "3.4.7"
      }
    });
```

Note that it's not needed to explicitly enable analytics by setting `urlAnalytics: true` when using  `trackedAnalytics`.

Let us know if you'd like to see a different implementation. We could also need some pointers on how to make this a complete PR, do we need to add types etc.

**Comment by colbyfayock** [comment]
note to CLD team - i'll create a followup ticket with Docs team to potentially add this though the URL Gen doesnt include Docs for this feature. separate discussion

**Comment by raae** [comment]
With the last commit, it seems to pass all the checks. I'll move this to "ready for review".

**Comment by raae** [comment]
Removed the `trackedAnalytics` configuration object and ensured all options fall back to the value returned by `sdkVersions()`.

Example usage as shown in the added test:
```
let imgStr = cloudinary.image("hello", {
  format: "png",
  urlAnalytics: true,
  sdkCode: "X",
  sdkSemver: "7.3.0",
  techVersion: "3.4.7"
});
```

**Review by cloudinary-pkoniu — APPROVED** [review]

File: test/unit/sdkAnalytics/imageTagWithAnalytics.spec.js [hunk 1]
```diff
@@ -65,4 +65,21 @@ describe('Tests for sdk analytics through image tag', function () {
...
+    let imgStr = cloudinary.image("hello", {
+      format: "png",
+      urlAnalytics: true,
+      sdkCode: "X",
+      sdkSemver: "7.3.0",
+      techVersion: "3.4.7"
+      
+    });
+
+    expect(imgStr).to.contain(`src='http://res.cloudinary.com/${TEST_CLOUD_NAME}/image/upload/hello.png?_s=AXAEzGT0`);
```
Comment (cloudinary-pkoniu) [inline test/unit/sdkAnalytics/imageTagWithAnalytics.spec.js:83]: Shouldn't we use `_a` instead of `_s`?

Comment (raae) [reply test/unit/sdkAnalytics/imageTagWithAnalytics.spec.js:83]: I'll go ahead and change it today.

Comment (raae) [reply test/unit/sdkAnalytics/imageTagWithAnalytics.spec.js]: Done!

**Pull request merged**